exporters/trace/jaeger/env_test.go

```diff
@@ -0,0 +1,398 @@
+
+package jaeger
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/kv/value"
+)
+
+func Test_parseTags(t *testing.T) {
+	require.NoError(t, os.Setenv("existing", "not-default"))
+
+	tags := "key=value,k1=${nonExisting:default}, k2=${withSpace:default},k3=${existing:default},k4=true,k5=42,k6=-1.2"
+	ts := parseTags(tags)
+	assert.Equal(t, 7, len(ts))
+
+	assert.Equal(t, kv.Key("key"), ts[0].Key)
+	assert.Equal(t, value.String("value"), ts[0].Value)
+
+	assert.Equal(t, kv.Key("k1"), ts[1].Key)
+	assert.Equal(t, value.String("default"), ts[1].Value)
+
+	assert.Equal(t, kv.Key("k2"), ts[2].Key)
+	assert.Equal(t, value.String("default"), ts[2].Value)
+
+	assert.Equal(t, kv.Key("k3"), ts[3].Key)
+	assert.Equal(t, value.String("not-default"), ts[3].Value)
+
+	assert.Equal(t, kv.Key("k4"), ts[4].Key)
+	assert.Equal(t, value.Bool(true), ts[4].Value)
+
+	assert.Equal(t, kv.Key("k5"), ts[5].Key)
+	assert.Equal(t, value.Int64(42), ts[5].Value)
+
+	assert.Equal(t, kv.Key("k6"), ts[6].Key)
+	assert.Equal(t, value.Float64(-1.2), ts[6].Value)
+
+	require.NoError(t, os.Unsetenv("existing"))
+}
+
+func Test_parseValue(t *testing.T) {
+	testCases := []struct {
+		name     string
+		str      string
+		expected value.Value
+	}{
+		{
+			name:     "bool: true",
+			str:      "true",
+			expected: value.Bool(true),
+		},
+		{
+			name:     "bool: false",
+			str:      "false",
+			expected: value.Bool(false),
+		},
+		{
+			name:     "int64: 012340",
+			str:      "012340",
+			expected: value.Int64(12340),
+		},
+		{
+			name:     "int64: -012340",
+			str:      "-012340",
+			expected: value.Int64(-12340),
+		},
+		{
+			name:     "int64: 0",
+			str:      "0",
+			expected: value.Int64(0),
+		},
+		{
+			name:     "float64: -0.1",
+			str:      "-0.1",
+			expected: value.Float64(-0.1),
+		},
+		{
+			name:     "float64: 00.001",
+			str:      "00.001",
+			expected: value.Float64(0.001),
+		},
+		{
+			name:     "float64: 1E23",
+			str:      "1E23",
+			expected: value.Float64(1e23),
+		},
+		{
+			name:     "string: foo",
+			str:      "foo",
+			expected: value.String("foo"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			v := parseValue(tc.str)
+			assert.Equal(t, tc.expected, v)
+		})
+	}
+}
+
+func TestNewRawExporterWithEnv(t *testing.T) {
+	const (
+		collectorEndpoint = "http://localhost"
+		username          = "user"
+		password          = "password"
+		serviceName       = "test-service"
+		disabled          = "false"
+		tags              = "key=value"
+	)
+
+	require.NoError(t, os.Setenv(envEndpoint, collectorEndpoint))
+	require.NoError(t, os.Setenv(envUser, username))
+	require.NoError(t, os.Setenv(envPassword, password))
+	require.NoError(t, os.Setenv(envDisabled, disabled))
+	require.NoError(t, os.Setenv(envServiceName, serviceName))
+	require.NoError(t, os.Setenv(envTags, tags))
```

Comment:
Just wondering whether for these tests which change the global environment whether it would be better to do a `LookupEnv()` first, and then restore that value at the end, rather than overwriting whatever may have been there and then unsetting the variable at the end of the function.

Reply:
You are right. That's a good idea.

I can write testing helper functions to modify and restore environment variables, and because these functions look like can be repeatedly used, so I like to put these helpers in `internal/testing` package.

Reply:
Yes, I think this is the first place I've seen where we pull things out of environment variables and also need tests for it. But I think there will be more - e.g. for Resources #785